components/ContributionConfirmationModal.js

```diff
@@ -0,0 +1,237 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { getCurrencySymbol } from '../lib/currency-utils';
+import { i18nGraphqlException } from '../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../lib/graphql/helpers';
+
+import Container from './Container';
+import FormattedMoneyAmount from './FormattedMoneyAmount';
+import { Box, Flex } from './Grid';
+import InputField from './InputField';
+import StyledButton from './StyledButton';
+import StyledHr from './StyledHr';
+import Modal, { ModalBody, ModalFooter, ModalHeader } from './StyledModal';
+import { P, Span } from './Text';
+import { TOAST_TYPE, useToasts } from './ToastProvider';
+
+const confirmContributionMutation = gqlV2/* GraphQL */ `
+  mutation ConfirmContribution($id: String!, $action: ProcessOrderAction!, $details: OrderDetailsInput) {
+    processPendingOrder(order: { id: $id }, action: $action, details: $details) {
+      id
+      status
+      permissions {
+        id
+        canMarkAsPaid
+        canMarkAsExpired
+      }
+      amount {
+        currency
+        valueInCents
+      }
+      platformContributionAmount {
+        currency
+        valueInCents
+      }
+    }
+  }
+`;
+
+const ContributionConfirmationModal = ({ order, onClose, show }) => {
+  const amountInitiated = order.amount.valueInCents + order.platformContributionAmount.valueInCents;
+  const currency = order.amount.currency;
+  const [amountReceived, setAmountReceived] = useState(amountInitiated);
+  const [platformTip, setPlatformTip] = useState(order.platformContributionAmount.valueInCents);
+  const [paymentProcessorFee, setPaymentProcessorFee] = useState(0);
+  const [netAmount, setNetAmount] = useState(amountReceived - platformTip);
+  const intl = useIntl();
+  const { addToast } = useToasts();
+  const [confirmOrder, { loading }] = useMutation(confirmContributionMutation, { context: API_V2_CONTEXT });
+
+  useEffect(() => {
+    setNetAmount(amountReceived - platformTip);
+  }, [amountReceived, platformTip]);
+
+  const triggerAction = async () => {
+    // Prevent submitting the action if another one is being submitted at the same time
+    if (loading) {
+      return;
+    }
+
+    const orderConfirmationDetails = {
+      totalAmount: { valueInCents: amountReceived, currency },
+      paymentProcessorFeesAmount: { valueInCents: paymentProcessorFee, currency },
+      platformTipAmount: { valueInCents: platformTip, currency },
+    };
```

Comment:
I asked Leo to change those values on the API side. Ideally I see:

```
amount: amountReceived + paymentProcessorFeesAmount - platformTipAmount,
paymentProcessorFeesAmount,
platformTipAmount
```

Reply:
These parts have been refactored to account for the new API changes. Now we are sending integer amounts not `Amount` objects back. Let me know if you see any further issues.

Reply:
This is wrong.

All amounts inputs should be `AmountInput` I don't understand why we're switching back to integers, this is backward.

Reply:
We have refactored the code both in API and Frontend to use `AmountInput`. 👍🏽